### ISSUE TITLE
Future Value Invalid Cast Exception - EF Core 1.1-preview1

### ISSUE THREAD

**Issue by saad749** [zzzprojects/EntityFramework-Plus#72]
Unable to cast object of type 'Microsoft.EntityFrameworkCore.Internal.LazyRef`1[Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IStateManager]' to type 'Microsoft.EntityFrameworkCore.ChangeTracking.Internal.IStateManager'.

I am getting this error after updating from 1.4.6 to 1.4.9 .

The normal EF count is working fine.

```
public QueryFutureValue<long> CountAllFuture()
        {
            return _dbContext.Set<T>().DeferredLongCount().FutureValue();
        }
```

Exception at line:
`var imCount = _imRepository.CountAllFuture();`

**Comment by saad749** [comment]
P.S. I also upgraded from EF Core 1.0 to EF Core 1.1 Preview 1.
Reverting back to 1.4.6 doesnt helps. 
and I am getting same errors for BatchDelete as well.

**Comment by zzzprojects** [comment]
Hello @saad749 ,

Thank you for reporting this issue.

Obviously, this issue happens because EF Core has again changed some stuff.

This issue is very easy to fix so it will be done this weekend but I cannot guarantee our library will be compatible since we don't know yet how much EF Core 1.1.0 has been changed.

I will let you know in a few days,

Best Regards,

Jonathan

**Comment by zzzprojects** [comment]
Hello @saad749 ,

The v1.4.10 is now available. 

Let me know if this is now compatible with EF Core 1.1 Preview 1

Best Regards,

Jonathan

**Comment by saad749** [comment]
Thanks a lot Jonathon for such a prompt fix. I will be testing it in 12-16 hours and let you know.

**Comment by zzzprojects** [comment]
Closing: Fixed in v1.4.10, if you still have the issue, re-open it.
